src/AnimalClub/ForeignBindings.hs

```diff
@@ -26,15 +27,41 @@ import           AnimalClub.Skellygen.Mesh
 
 import           Data.Convertible
 import           Data.Int                     (Int32)
+import qualified Data.Text                    as T
 import qualified Data.Vector.Storable         as V
 import qualified Data.Vector.Storable.Mutable as MV
 import           Foreign
 import           Foreign.C.Types
---import           Foreign.Storable.Tuple
+import           Foreign.Storable             (Storable (..))
+import qualified Foreign.Storable.Record      as Store
 import           System.Random
 
 import           Debug.Trace
 
+
+instance (Storable a, Storable b, Storable c, Storable d, Storable e, Storable f, Storable g, Storable h) => Storable (a,b,c,d,e,f,g,h) where
+   sizeOf    = Store.sizeOf storeHextuple
+   alignment = Store.alignment storeHextuple
+   peek      = Store.peek storeHextuple
+   poke      = Store.poke storeHextuple
+
+{-# INLINE storeHextuple #-}
+storeHextuple ::
+  (Storable a, Storable b, Storable c, Storable d, Storable e, Storable f, Storable g, Storable h) =>
+  Store.Dictionary (a,b,c,d,e,f,g,h)
+storeHextuple =
+   Store.run $
+   pure (,,,,,,,)
+      <*> (Store.element $ \(x,_,_,_,_,_,_,_) -> x)
```

Comment:
Could you use a function like `put_elem` in elixir with an index so it's easier to update in the future? Matching is probably faster though

Reply:
haskell has some tuple helpers but they only go up to like 3-tuples :(